### PR TITLE
docs: update sudo configuration to a more restrictive variant

### LINKED ISSUE #993: docs: sudo permissions too extensive
Currently the docs of the provider say that you should add

```
terraform ALL=(ALL) NOPASSWD:ALL
```

to the sudoers file.

Looking at the code and trying myself it suffices to add something like this to the sudoers file:

```
terraform ALL=(root) NOPASSWD: /sbin/pvesm
terraform ALL=(root) NOPASSWD: /sbin/qm
terraform ALL=(root) NOPASSWD: /usr/bin/echo tfpve
```

with the principle of least privilege.

By changing the "is sudo installed"-check we could even get rid of the echo-line.

### PR DESCRIPTION
**Pull request by Eusebius1920**
### Contributor's Note

- [x] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated templates in `/example` for any new or updated resources / data sources.
- [ ] I have ran `make example` to verify that the change works as expected.

### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #993

### CODE REVIEW

**Comment by Eusebius1920** [comment]
I updated the docs. If I find some spare time next week, I will try to change the `try_sudo` variant to get rid of the third sudo rule...

File: docs/index.md [hunk 1]
```diff
@@ -177,10 +179,10 @@ You can configure the `sudo` privilege for the user via the command line on the
 - Test the SSH connection and password-less `sudo`:
   
     ```sh
-    ssh terraform@<target-node> sudo ls -la /root
+    ssh terraform@<target-node> sudo echo tfpve
```
Comment (bpg-dev) [inline docs/index.md:182]: Technically, if user is in sudoers, but is not configured to run as `root`, the `sudo echo tfpve` will still work, but nothing else that requires root access. 

But that's fine for now, assuming the check will be updated soon.

**Review by bpg — APPROVED** [review]
Thanks for the PR @Eusebius1920!

LGTM! 🚀

**Comment by bpg** [comment]
@all-contributors please add @Eusebius1920 for bug, docs

**Comment by bpg** [comment]
> I updated the docs. If I find some spare time next week, I will try to change the `try_sudo` variant to get rid of the third sudo rule...

@Eusebius1920 I've added that in #1004

**Pull request merged**